.github/workflows/gtest-bare-metal.yml

```diff
@@ -161,3 +143,11 @@ jobs:
       - name: Run st2110 st20p test case with kernel loopback
         run: |
           ./build/tests/KahawaiTest --p_port kernel:lo --r_port kernel:lo --auto_start_stop --gtest_filter=St20p*
+
+      - name: Kill gtest routines that could have been left
+        if: always()
+        run: |
+          sudo killall -SIGINT KahawaiTest || true
+          sudo killall -SIGINT KahawaiUfdTest || true
+          sudo killall -SIGINT KahawaiUplTest || true
+          sudo killall -SIGINT MtlManager || true
```

Comment:
Is it possible to return error in case of routines kill, for example:
1. Return 0 if there is no processes left -> PASS
2. Return 1 if at least one process has been killed -> FAIL

Reply:
After short PM talk a case of deadlocked unit tests was proposed by @tszumski - and the question itself was to a purpose of killing processes before/after the workflow and if it won't introduce any false positive results.

Currently, as this only a cleaning case before/after execution on self-hosted bare metal - it should be treated as preventive clean-up - but still a valid point to consider in the future.